src/linter/block.go

```diff
@@ -500,6 +498,33 @@ func (b *BlockWalker) EnterNode(w walker.Walkable) (res bool) {
 	return res
 }
 
+func (b *BlockWalker) handleInterface(int *stmt.Interface) bool {
+	for _, st := range int.Stmts {
+		switch x := st.(type) {
+		case *stmt.ClassMethod:
+			for _, modifier := range x.Modifiers {
+				if strings.ToLower(modifier.Value) != "public" {
+					interfaceName := int.InterfaceName.Value
+					methodName := x.MethodName.Value
+					b.r.Report(x, LevelWarning, "nonPublicInterfaceMethod", "Non-public method '%s' in the interface '%s'", methodName, interfaceName)
```

Comment:
This diagnostic name in combination with the warning text will result in a stutter:
```
nonPublicInterfaceMethod: Non-public constant 'b' in the interface 'A'
```
It's better to avoid that. We say "non public in interface" twice on the same line.
It may be clearer this way:
```
nonPublicInterfaceMethod: 'b' can't be private
```
The diagnostic name reads like a message prefix, so it's redundant to repeat it.

I also suggest giving both warnings as a single diagnostic, `nonPublicInterfaceMember`.